### ISSUE TITLE
Add term and year to course titles in collection UI to disambiguate courses with the same title

### ISSUE THREAD

**Issue by noisecapella** [mitodl/ocw-studio#393]
Related to #391

We should disambiguate courses so we don't see duplicate listings, perhaps by adding a date to the title:

![Screenshot from 2021-07-16 15-56-09](https://user-images.githubusercontent.com/863262/126004008-f7eafea7-c72b-48ee-8432-1a19e7e51771.png)

**Comment by pdpinch** [comment]
The OCW team is used to disambiguating courses using the term and the year.

**Comment by ChristopherChudzicki** [comment]
Do we want to use short ids here instead of course+year, like in https://github.com/mitodl/ocw-studio/issues/1265 ?

**Comment by pdpinch** [comment]
Good idea. Will it fit?

**Comment by ChristopherChudzicki** [comment]
I think so; we'll see. @pdpinch did you mean to move this out of to-do?

**Comment by pdpinch** [comment]
I did, just because I want to focus on publishing blockers today (Friday). 

It’s OK to work on this, it’s just low priority.

**Comment by MAbdurrehman12** [comment]
@ChristopherChudzicki Are you working on this?

**Comment by ChristopherChudzicki** [comment]
I am not. Thanks for checking... I'll unassign myself from this and other stuff for the moment.

**Comment by ChristopherChudzicki** [comment]
I've been reviewing @MAbdurrehman12 PR for this (https://github.com/mitodl/ocw-studio/pull/1375). This turns out to be kind of hard to do well in OCW right now. Right now, it's feasible to get the short id into the "select" dropdown (since that's populated by OCW studio), but rather hard to get them in the sortable list:

<img width="1097" alt="Screenshot 2023-03-03 at 10 55 53 AM" src="https://user-images.githubusercontent.com/9010790/222772210-0dda2ac7-e8d5-4e27-a131-7d22e365b7c9.png">

The sortable area is populated by course list resource metadata. Fundamentally, I think our course lists aren't storing the right data.

WebsiteContent course list metadata look like this:
```yaml
content_type: course-lists
courses:
- id: courses/11-165j-urban-energy-systems-and-policy-fall-2021
  title: Urban Energy Systems and Policy
- id: courses/4-401-environmental-technologies-in-buildings-fall-2018
  title: Environmental Technologies in Buildings
# ...
```

- `id` is really `website.url_path`. This is used by Hugo to fetch website data from the static `data.json` files. (E.g., for making course cards in course lists).
-It is only used as a label for the items in OCW Studio. *Theoretically hugo could use this, but it does not. It's also not great...if someone changes website Title, the value here will be out of date.*

Ideally, it seems to me that the course lists would provide an identifier that Hugo can use (which we have `url_path`) and either an identifier Studio can use to fetch the relevant information (e.g., website.uuid) or the studio API could return extra information when fetching a course list.

**Comment by MAbdurrehman12** [comment]
@ChristopherChudzicki @pdpinch FYI: Had a call with @HussainTaj-arbisoft, we discussed the ticket and the work going on in the PR along with the issues in it. I'm gonna this ticket to Hussain and he can continue working on it from here.

**Comment by HussainTaj-arbisoft** [comment]
https://github.com/mitodl/ocw-studio/pull/1715 will be reverted because short_id is not showing up on search results.

**Comment by pdpinch** [comment]
@HussainTaj-arbisoft I'm confused. Can you summarize what happened with #1375 and #1721 ?

What to do you think we need to do to get this to work? If it helps, it's not strictly necessary that we use the `short_id` -- we just need to display some data to differentiate courses with similar names. A term and year would be just as good. Even a year would help.

**Comment by HussainTaj-arbisoft** [comment]
@pdpinch 

> Can you summarize what happened with #1375 and #1721 ?

The approach in https://github.com/mitodl/ocw-studio/pull/1375 made it challenging to sync two lists in the code. The details are in this comment https://github.com/mitodl/ocw-studio/pull/1375#discussion_r1124661620

https://github.com/mitodl/ocw-studio/pull/1721 was a rewrite of the same solution with a different approach. When I tested it on RC the short_id did not appear in the search results. It was my bad, I probably missed the part that renders the search results.

> What to do you think we need to do to get this to work? 

 I'll take a look at this again next week. I'm hoping there will be a small change to https://github.com/mitodl/ocw-studio/pull/1721 and that should make it work.
